### ISSUE TITLE
BetterLoadingScreen startup time

### ISSUE THREAD

**Issue by Methes** [GTNewHorizons/GT-New-Horizons-Modpack#6333]
#### Which modpack version are you using?
2.0.9.0 QF4
#
After updating the pack from 2.0.8.8 I noticed my client startup time increased severely. Especially the baking phase regularly took 400s which was unheard of for my PC before updating.

After removing the BetterLoadingScreen my baking phase takes 55s again as it used to.

Something is seriously wrong with that mod, at least on my PC.

**Comment by Dream-Master** [comment]
If you removed the mod it will work. I cant add two pack options in technical manual removal is necessary.

**Comment by Methes** [comment]
Sure, I can remove it and I did.
I was more looking for startup times with and without the mod of other people, cause if it does increase startup times 8(!) times for others as well, the mod should be removed altogether.

**Comment by Prometheus0000** [comment]
If this really increases loading time that much, why do we have it in? We should remove it until/unless someone fixes that. Also see:  #6125

**Comment by Dream-Master** [comment]
@jackowski626 any idea ?

**Comment by jackowski626** [comment]
I did tests on my machine, including an external profiler. On my new it goes even faster. Maybe someone else could do profiling too? Or just looking at the logs, the second time the sound manager thing is loaded Mc is up. So measure from start to that time. Can you disable changing backgrounds and test? And if it doesn't help, enable "salt mode" in config. If salt mode doesn't help, then it's something deeper. Also, what are your specs

**Comment by Prometheus0000** [comment]
When I actually tried removing it, it didn't seem to change the loading time significantly, if at all. I didn't exactly time it, or do that other stuff though. @Methes

**Comment by jackowski626** [comment]
Look at logs

**Comment by Prometheus0000** [comment]
I can upload logs, but I need to know which one.

**Comment by jackowski626** [comment]
Iirc latest log has this info. Just need latest log from a launch with and without bls

**Comment by Prometheus0000** [comment]
With
[fml-client-latest.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/5433771/fml-client-latest.log)
Without
[fml-client-latest.log](https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/files/5433776/fml-client-latest.log)
It's about 5.5m either way

**Comment by Methes** [comment]
As I said before, Baking phase takes the biggest hit, reaching several minutes to complete.
5:29 without BLS
11:06 with BLS
11:09 with backgroundChanging=false
9:48 with backgroundChanging=false and salt=true

i7-4790K
16GB RAM
Running on SSD

**Comment by jackowski626** [comment]
Hmm 10 minutes while every feature of bls is basically turned off and 5:30 when bls gone. That's weird. Maybe you have some odd optimization mod? Or some conflicting OptiFine settings? Or java arguments?

**Comment by Methes** [comment]
BetterFoliage, Fastcraft 1.25, Optifine U_E7
4GB max memory allocation
-d64 -XX:+UseConcMarkSweepGC -XX:ParallelGCThreads=4 -XX:+DisableExplicitGC

**Comment by jackowski626** [comment]
Betterfoliage and of should be fine, unless of has some weird settings. Maybe it's fastcraft?

**Comment by Methes** [comment]
I'll try tomorrow with vanilla GTNH and no java args.

**Comment by GTNH-Afx237v7** [comment]
I can confirm that there is a difference with BLS enabled or not. Usually baking takes around 180-200 seconds for me. Total load time (from Launch in MultiMC to seeing the main menu) is 10:15min.
With BLS disabled (and splash.properties set to disabled) it was 8:35min, while baking took about 80 seconds.

**Comment by boubou19** [comment]
On my side, when I use BLS, I need 7 min to boot, when I don't use it, I need 6 minute to boot.

**Comment by jackowski626** [comment]
Could you tell me the time when you disable changing backgrounds?

**Comment by Glease** [comment]
Whoever get slow down please try remove optifine and see if the slow down go away. I'm not asking you to play without optifine and all its fancy features. I just want you to kindly help us find the cause of this problem.

**Comment by Prometheus0000** [comment]
Also note that mitch has sped up loading times since before, though that could be on world load and not to main screen load.

**Comment by GTNH-Afx237v7** [comment]
The optimization to Minetweaker from mitch speed up world loading. Helped me go from 80ish seconds to about 20 seconds.

**Comment by q009** [comment]
I noticed a similar situation when using Optifine.
Normally, the game starts in just over 3 minutes for me. I added Optifine and it took over 9 minutes to start. I can confirm that it's mostly the GT materials registration step taking so much longer.
Having removed Better Loading Screen, the game loaded in well under 2 minutes.
There's definitely something funny happening between Optifine and BLS, I saw this issue reported: https://github.com/sp614x/optifine/issues/304
While it's for a newer version, it might still be related, because the rootcause seems to be around vsync and splash screen handling.
